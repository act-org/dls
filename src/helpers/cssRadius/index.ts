/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

import { StandardCSSProperties } from '@mui/system';

import px from '~/helpers/px';

type BorderRadius = Pick<
  StandardCSSProperties,
  | 'borderBottomLeftRadius'
  | 'borderBottomRightRadius'
  | 'borderRadius'
  | 'borderTopLeftRadius'
  | 'borderTopRightRadius'
>;

export const cssRadius = (value: number): BorderRadius => ({
  borderBottomLeftRadius: px(value),
  borderBottomRightRadius: px(value),
  borderRadius: px(value),
  borderTopLeftRadius: px(value),
  borderTopRightRadius: px(value),
});

export default cssRadius;
