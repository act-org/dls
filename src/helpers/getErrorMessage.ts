/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { get } from 'lodash';

export const getErrorMessage = (err: unknown): string => {
  let errorMessage = '';
  const graphQlErrorMessage = get(err, 'networkError.result.errors[0].message');
  const standardErrorMessage = get(err, 'message');

  if (graphQlErrorMessage) {
    errorMessage = graphQlErrorMessage;
  } else if (standardErrorMessage) {
    errorMessage = standardErrorMessage;
  }

  return errorMessage.trim();
};

export default getErrorMessage;
