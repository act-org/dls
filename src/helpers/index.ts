/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

export * from './cssRadius';
export * from './getErrorMessage';
export * from './makeShadow';
export * from './mergeClasses';
export * from './oxfordCommaJoin';
export * from './px';
export * from './pxToNumber';
export * from './search';
export * from './sort';
export * from './styled';
export * from './types';
