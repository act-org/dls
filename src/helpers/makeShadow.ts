/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

import px from '@actinc/dls/helpers/px';

export const makeShadow = (
  xOffset: number,
  yOffset: number,
  blurRadius: number | undefined,
  spreadRadius: number | undefined,
  color: string,
  inset = false,
): string => {
  return `${inset ? 'inset ' : ''}${px(xOffset)} ${px(yOffset)} ${`${px(
    blurRadius,
  )} `}${`${px(spreadRadius)} `}${color}`;
};

export default makeShadow;
