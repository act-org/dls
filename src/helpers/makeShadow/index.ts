/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import isNumber from 'lodash/isNumber';

import px from '~/helpers/px';

export const makeShadow = (
  xOffset: number,
  yOffset: number,
  blurRadius: number | undefined,
  spreadRadius: number | undefined,
  color: string,
  inset = false,
): string => {
  let str = '';

  if (inset) {
    str = `${str} inset`;
  }

  if (isNumber(xOffset)) {
    str = `${str} ${px(xOffset)}`;
  }

  if (isNumber(yOffset)) {
    str = `${str} ${px(yOffset)}`;
  }

  if (isNumber(blurRadius)) {
    str = `${str} ${px(blurRadius)}`;
  }

  if (isNumber(spreadRadius)) {
    str = `${str} ${px(spreadRadius)}`;
  }

  if (color) {
    str = `${str} ${color}`;
  }

  return str.trim();
};

export default makeShadow;
