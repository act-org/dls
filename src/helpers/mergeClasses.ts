/**
 * Copyright (c) Encoura, LLC and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import clsx from 'clsx';
import mergeWith from 'lodash/mergeWith';

type Classes = any;

export const mergeClasses = (classes1: Classes, classes2: Classes): Classes =>
  mergeWith(classes1 || {}, classes2 || {}, (className1, className2): string =>
    clsx(className1, className2),
  );

export default mergeClasses;
