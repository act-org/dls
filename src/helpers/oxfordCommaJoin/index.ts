/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

export const oxfordCommaJoin = (input: string[]): string => {
  const clone = [...input];

  if (input.length <= 2) {
    return clone.join(' and ');
  }

  const last = clone.pop();
  return `${clone.join(', ')}, and ${last}`;
};

export default oxfordCommaJoin;
