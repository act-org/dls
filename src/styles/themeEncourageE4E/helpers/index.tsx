/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

export type ColorIndex =
  | 50
  | 100
  | 200
  | 300
  | 400
  | 500
  | 600
  | 700
  | 800
  | 900
  | string;

interface IBrightnessMap {
  brightness: number;
  colorIndex: ColorIndex;
}

/*
 * Do not modify these values unless UX give us new values,
 * they are used to generate our Figma palette
 */
const brightnessCoefficient: IBrightnessMap[] = [
  {
    brightness: 0.9,
    colorIndex: 50,
  },
  {
    brightness: 0.69,
    colorIndex: 100,
  },
  {
    brightness: 0.54,
    colorIndex: 200,
  },
  {
    brightness: 0.33,
    colorIndex: 300,
  },
  {
    brightness: 0.2,
    colorIndex: 400,
  },
  {
    brightness: 0,
    colorIndex: 500,
  },
  {
    brightness: -0.09,
    colorIndex: 600,
  },
  {
    brightness: -0.29,
    colorIndex: 700,
  },
  {
    brightness: -0.45,
    colorIndex: 800,
  },
  {
    brightness: -0.58,
    colorIndex: 900,
  },
];

export const hexToRGB = (color: string): number[] | undefined => {
  let r = '';
  let g = '';
  let b = '';

  if ((color.length !== 4 && color.length !== 7) || color[0] !== '#')
    return undefined;

  if (color.length === 4) {
    r = `0x${color[1]}${color[1]}`;
    g = `0x${color[2]}${color[2]}`;
    b = `0x${color[3]}${color[3]}`;
  } else if (color.length === 7) {
    r = `0x${color[1]}${color[2]}`;
    g = `0x${color[3]}${color[4]}`;
    b = `0x${color[5]}${color[6]}`;
  }

  return [Number(r), Number(g), Number(b)];
};

export const RGBToHex = (r: number, g: number, b: number): string => {
  let R = r.toString(16);
  let G = g.toString(16);
  let B = b.toString(16);

  if (R.length === 1) R = `0${R}`;
  if (G.length === 1) G = `0${G}`;
  if (B.length === 1) B = `0${B}`;

  return `#${R.slice(0, 2)}${G.slice(0, 2)}${B.slice(0, 2)}`;
};

export const lightenRGB = (
  color: string,
  percentage: number,
): string | undefined => {
  const RGBColor = hexToRGB(color);

  if (!RGBColor) return undefined;

  if (percentage === 0) return color;

  if (percentage > 0) {
    return RGBToHex(
      Math.round(RGBColor[0] + ((255 - RGBColor[0]) * percentage) / 100),
      Math.round(RGBColor[1] + ((255 - RGBColor[1]) * percentage) / 100),
      Math.round(RGBColor[2] + ((255 - RGBColor[2]) * percentage) / 100),
    );
  }

  return RGBToHex(
    Math.round(RGBColor[0] * (1 + percentage / 100)),
    Math.round(RGBColor[1] * (1 + percentage / 100)),
    Math.round(RGBColor[2] * (1 + percentage / 100)),
  );
};

export const figmaColorGenerator = (
  color: string,
): Record<ColorIndex, string> => {
  const colorPalette = {} as unknown as Record<ColorIndex, string>;
  brightnessCoefficient.forEach(({ brightness, colorIndex }) => {
    const generatedColor = lightenRGB(color, brightness * 100);
    if (generatedColor) colorPalette[colorIndex] = generatedColor;
  });
  return colorPalette;
};

export const generatePaletteColors = <
  T extends { colorName: U; colorHEX: string },
  U extends string,
>(
  colors: T[],
): Record<U, Record<ColorIndex, string>> => {
  const colorPalette = {} as unknown as Record<U, Record<ColorIndex, string>>;
  colors.forEach(color => {
    colorPalette[color.colorName] = figmaColorGenerator(color.colorHEX);
  });
  return colorPalette;
};
