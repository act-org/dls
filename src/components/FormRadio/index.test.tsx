/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

import { standard } from '@actinc/dls/helpers/test';

import { FormRadio } from '.';

describe('FormRadio', () => {
  const Component = <FormRadio id="primaryswitch" label="The label" />;
  standard(Component);
});
