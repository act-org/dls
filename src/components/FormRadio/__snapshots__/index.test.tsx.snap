// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormRadio ACT theme matches the snapshot 1`] = `
<div>
  <label
    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-9ttzdx-MuiFormControlLabel-root"
    id="primaryswitch"
  >
    <span
      class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-13qwt2c-MuiButtonBase-root-MuiRadio-root"
      color="primary"
    >
      <input
        class="PrivateSwitchBase-input css-j8yymo"
        type="radio"
        value=""
      />
      <span
        class="css-1qiat4j"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-a4fljk-MuiSvgIcon-root"
          data-testid="RadioButtonUncheckedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1jyczwl-MuiSvgIcon-root"
          data-testid="RadioButtonCheckedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
          />
        </svg>
      </span>
    </span>
    <span
      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-1or18zx-MuiTypography-root"
    >
      The label
    </span>
  </label>
</div>
`;

exports[`FormRadio ACT_ET theme matches the snapshot 1`] = `
<div>
  <label
    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1ijgvmu-MuiFormControlLabel-root"
    id="primaryswitch"
  >
    <span
      class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-oeumqa-MuiButtonBase-root-MuiRadio-root"
    >
      <input
        class="PrivateSwitchBase-input css-j8yymo"
        type="radio"
        value=""
      />
      <span
        class="css-1qiat4j"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1pd3nvo-MuiSvgIcon-root"
          data-testid="RadioButtonUncheckedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-15jvua8-MuiSvgIcon-root"
          data-testid="RadioButtonCheckedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
          />
        </svg>
      </span>
    </span>
    <span
      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-uuxgud-MuiTypography-root"
    >
      The label
    </span>
  </label>
</div>
`;

exports[`FormRadio ENCOURA theme matches the snapshot 1`] = `
<div>
  <label
    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-9ttzdx-MuiFormControlLabel-root"
    id="primaryswitch"
  >
    <span
      class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-1vhsnjs-MuiButtonBase-root-MuiRadio-root"
    >
      <input
        class="PrivateSwitchBase-input css-j8yymo"
        type="radio"
        value=""
      />
      <span
        class="css-1qiat4j"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-pejxdp-MuiSvgIcon-root"
          data-testid="RadioButtonUncheckedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-14lsiqu-MuiSvgIcon-root"
          data-testid="RadioButtonCheckedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
          />
        </svg>
      </span>
    </span>
    <span
      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-vt4fc5-MuiTypography-root"
    >
      The label
    </span>
  </label>
</div>
`;

exports[`FormRadio ENCOURA_CLASSIC theme matches the snapshot 1`] = `
<div>
  <label
    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1ijgvmu-MuiFormControlLabel-root"
    id="primaryswitch"
  >
    <span
      class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-y1gler-MuiButtonBase-root-MuiRadio-root"
    >
      <input
        class="PrivateSwitchBase-input css-j8yymo"
        type="radio"
        value=""
      />
      <span
        class="css-1qiat4j"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-pejxdp-MuiSvgIcon-root"
          data-testid="RadioButtonUncheckedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-14lsiqu-MuiSvgIcon-root"
          data-testid="RadioButtonCheckedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
          />
        </svg>
      </span>
    </span>
    <span
      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-vt4fc5-MuiTypography-root"
    >
      The label
    </span>
  </label>
</div>
`;

exports[`FormRadio ENCOURAGE theme matches the snapshot 1`] = `
<div>
  <label
    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-11n2pmj-MuiFormControlLabel-root"
    id="primaryswitch"
  >
    <span
      class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-1qulgto-MuiButtonBase-root-MuiRadio-root"
    >
      <input
        class="PrivateSwitchBase-input css-j8yymo"
        type="radio"
        value=""
      />
      <span
        class="css-1qiat4j"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1pd3nvo-MuiSvgIcon-root"
          data-testid="RadioButtonUncheckedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-15jvua8-MuiSvgIcon-root"
          data-testid="RadioButtonCheckedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
          />
        </svg>
      </span>
    </span>
    <span
      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-1c0jcfm-MuiTypography-root"
    >
      The label
    </span>
  </label>
</div>
`;

exports[`FormRadio ENCOURAGE_E4E theme matches the snapshot 1`] = `
<div>
  <label
    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1hu42yv-MuiFormControlLabel-root"
    id="primaryswitch"
  >
    <span
      class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-1vxgrnp-MuiButtonBase-root-MuiRadio-root"
    >
      <input
        class="PrivateSwitchBase-input css-j8yymo"
        type="radio"
        value=""
      />
      <span
        class="css-1qiat4j"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-tptydz-MuiSvgIcon-root"
          data-testid="RadioButtonUncheckedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1xvuveb-MuiSvgIcon-root"
          data-testid="RadioButtonCheckedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
          />
        </svg>
      </span>
    </span>
    <span
      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-1hvcons-MuiTypography-root"
    >
      The label
    </span>
  </label>
</div>
`;
